packages/server/src/api/controllers/row/alias.ts

```diff
@@ -22,7 +28,13 @@ export default class AliasTables {
     const char = this.character
     this.aliases[tableName] = char
     this.tableAliases[char] = tableName
-    this.character = String.fromCharCode(char.charCodeAt(0) + 1)
+    this.character =
+      char.substring(0, char.length - 1) +
+      String.fromCharCode(char.charCodeAt(char.length - 1) + 1)
+    // reached end of characters, extend number of characters used
+    if (this.character.charAt(this.character.length - 1) === "z") {
+      this.character = new Array(this.character.length + 1).fill("a").join("")
+    }
```

Comment:
This isn't quite correct. It'll never increase the earlier letters past `"a"`. I had a play with this in a script and came up with this:

```javascript
class CharSequence {
  static alphabet = "abcdefghijklmnopqrstuvwxyz";

  constructor() {
    this.counters = [0];
  }

  get character() {
    return this.counters.map((i) => CharSequence.alphabet[i]).join("");
  }

  next() {
    for (let i = this.counters.length - 1; i >= 0; i--) {
      if (this.counters[i] < CharSequence.alphabet.length - 1) {
        this.counters[i]++;
        return;
      }
      this.counters[i] = 0;
    }
    this.counters.unshift(0);
  }
}

const seq = new CharSequence();

for (let i = 0; i < 100; i++) {
  seq.next();
  console.log(seq.character);
}
```

Can extend the alphabet if we want to, to ensure maximum smallness on the aliases. For example, we might want to include capital letters as well. I believe SQL treats them as distinct to lowercase.

Reply:
Ah good catch - was still playing around with this when I logged off last night! Preference is to keep it to lowercase, just to avoid ambiguity, I think this is a pretty big edge case, as if a query includes minimum 26 different tables, its already going to be problematic for Budibase to deal with (performance wise).